File: backend/src/app/models/listing.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Float, ForeignKey, Numeric
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

import datetime

from app.core.database import Base

class Listing(Base):
    __tablename__ = "listings"

    id = Column(Integer, primary_key=True, index=True)
    host_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String(120), nullable=False)
    description = Column(String(1000), nullable=False)
    price = Column(Float, nullable=False)
    location = Column(String(255), nullable=False)
    room_type = Column(String(30), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now())
    sqft = Column(Integer, nullable=True)
    start_date = Column(DateTime(timezone=True), nullable=True)
    end_date = Column(DateTime(timezone=True), nullable=True)
    college_id = Column(Integer, nullable=True)
    thumbnail_url = Column(String, nullable=True, default=None)
    latitude = Column(Numeric(9,6), nullable=False)
    longitude = Column(Numeric(9,6), nullable=False)

    user = relationship("User", back_populates="listings")
    conversations = relationship("Conversation", back_populates="listing")

File: frontend/src/types/listing.ts
export interface Listing {
	id: number;
	host_id: number;
	title: string;
	description: string;
	price: number;
	location: string;
	room_type?: string | null;
	sqft?: number | null;
	start_date: string;
	end_date: string;
	college_id?: number | null;
	thumbnail_url?: string | null;
	latitude: number;
	longitude: number;
	created_at: string;
	updated_at: string;
}

/** One row from `GET /api/v1/listings/` (differs from full `Listing`). */
export interface ListingListApiRow {
	id: number;
	title: string;
	price: number;
	college: number | null;
	location_text: string;
	room_type?: string | null;
	sqft?: number | null;
	start_date: string | null;
	end_date: string | null;
	thumbnail_url?: string | null;
	created_at: string | null;
}
